src/evo/cbtx.cpp

```diff
@@ -314,8 +322,139 @@ bool CalcCbTxMerkleRootQuorums(const CBlock& block, const CBlockIndex* pindexPre
     return true;
 }
 
+bool CheckCbTxBestChainlock(const CBlock& block, const CBlockIndex* pindexPrev, const llmq::CChainLocksHandler& chainlock_handler, BlockValidationState& state)
+{
+    if (block.vtx[0]->nType != TRANSACTION_COINBASE) {
+        return true;
+    }
```

Comment:
this should never be true

Reply:
Yes but this check is present also in `CheckCbTxMerkleRoots`